### PR TITLE
fix: code signin was not returning primary account!

### PR DESCRIPTION
**Pull request by danielgent**
Wow, this PR was meant to be adding more silent auth checks. I did, and I found a bug :see_no_evil:

### CODE REVIEW

File: integration-test/flows/code-flow.spec.ts [hunk 1]
```diff
@@ -44,7 +44,8 @@ describe("code-flow", () => {
         client_id: "clientId",
         connection: "email",
         email: "test@example.com",
-        send: "link",
+        // can be code or link
+        send: "code",
```
Comment (danielgent) [inline integration-test/flows/code-flow.spec.ts:48]: May as well have this somewhere in our tests!

File: integration-test/flows/code-flow.spec.ts [hunk 2]
```diff
@@ -352,13 +351,28 @@ describe("code-flow", () => {
     );
 
     const accessToken2 = parseJwt(
-      new URL(tokenResponse.headers.get("location")!).searchParams.get(
+      new URL(tokenResponse2.headers.get("location")!).searchParams.get(
```
Comment (danielgent) [inline integration-test/flows/code-flow.spec.ts:354]: ooops! :see_no_evil: 

This was an issue. I was using a previous response...

Maybe these tests could be improved with more explicit naming... or scoping each part. Comments welcome!

Comment (danielgent) [reply integration-test/flows/code-flow.spec.ts:354]: This didn't actually spot the bug though. We wer still returning the correct `sub` in the `access_token`

File: integration-test/flows/code-flow.spec.ts [hunk 3]
```diff
@@ -352,13 +351,28 @@ describe("code-flow", () => {
...
+    const setCookiesHeader2 = tokenResponse2.headers.get("set-cookie")!;
+    const { idToken: silentAuthIdTokenPayload2 } =
+      await doSilentAuthRequestAndReturnTokens(
+        setCookiesHeader2,
+        worker,
+        nonce,
+        "clientId",
+      );
+    // second time round make sure we get the primary userid again
+    expect(silentAuthIdTokenPayload2.sub).toBe("userId");
```
Comment (danielgent) [inline integration-test/flows/code-flow.spec.ts:375]: :exploding_head:  this was returning the linked user

File: src/authentication-flows/ticket.ts [hunk 4]
```diff
@@ -40,7 +40,11 @@ export async function ticketAuth(
     ticket.email,
   );
 
-  let user = usersWithSameEmail.find((u) => u.provider === provider);
+  let user = usersWithSameEmail.find((u) => u.provider === provider) || null;
+
+  if (user?.linked_to) {
+    user = await env.data.users.get(tenant_id, user.linked_to);
+  }
```
Comment (danielgent) [inline src/authentication-flows/ticket.ts:47]: Ooops! Similar code exists on the `social` flow :facepalm:

File: integration-test/flows/code-flow.spec.ts (hunk 2 above)
Comment (danielgent) [reply integration-test/flows/code-flow.spec.ts:354]: Changing this actually exposed a bug (see below), as in this run the secondary user is already creating which we are signing in with

File: src/authentication-flows/ticket.ts (hunk 4 above)
Comment (danielgent) [reply src/authentication-flows/ticket.ts:47]: This is the bug mentioned above

File: integration-test/flows/social.spec.ts [hunk 5]
```diff
@@ -155,7 +155,30 @@ describe("social sign on", () => {
...
+          setCookiesHeader,
+          worker,
+          "nonce",
+          "clientId",
+        );
+        expect(silentAuthIdTokenPayload).toMatchObject({
+          sub: "demo-social-provider|123456789012345678901",
+          aud: "clientId",
+          name: "john.doe@example.com",
+        });
```
Comment (danielgent) [inline integration-test/flows/social.spec.ts:177]: some more silent auth calls. nothing to see here all good!

File: integration-test/flows/code-flow.spec.ts (hunk 2 above)
Comment (danielgent) [reply integration-test/flows/code-flow.spec.ts:354]: fixed below on `src/authentication-flows/ticket.ts`

**Review by markusahlstrand — APPROVED** [review]
LGTM!

**Pull request merged**